ASSERTION FAILED: containingBlock.isInFlowPositioned() while running `the-anchor-attribute-003-crash.tentative.html` WPT Test

obyknovenius opened the pull request:
#### 6a53726378e73e552515fcf134ddfba8a7c2eb1d
<pre>
ASSERTION FAILED: containingBlock.isInFlowPositioned() while running `the-anchor-attribute-003-crash.tentative.html` WPT Test
<a href="https://bugs.webkit.org/show_bug.cgi?id=267993">https://bugs.webkit.org/show_bug.cgi?id=267993</a>

Reviewed by NOBODY (OOPS!).

According to the spec [1], some properties, such as &quot;position&quot;,
&quot;transform&quot;, &quot;will-change&quot;, &quot;contain&quot;, can cause a box to establish an
absolute positioning containing block and become a containing block for
an absolute positioned box.

It makes this assertion invalid in some cases. We should remove it.

The similar change was done in Blink in [2] where this test was
introduced.

1. <a href="https://drafts.csswg.org/css-position/#absolute-positioning-containing-block">https://drafts.csswg.org/css-position/#absolute-positioning-containing-block</a>
2. <a href="https://bugs.chromium.org/p/chromium/issues/detail?id=1486148">https://bugs.chromium.org/p/chromium/issues/detail?id=1486148</a>

* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::containingBlockLogicalHeightForPositioned const):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6a53726378e73e552515fcf134ddfba8a7c2eb1d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/37479 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/16363 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/39742 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/40005 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/33401 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/38790 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/18979 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/13525 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31804 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/38044 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/13800 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/32883 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/12007 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/11991 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/33567 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/41268 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/33884 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/33988 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37888 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/12551 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/10471 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/36045 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/14012 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/12958 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/13327 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay reviewed (commented):
>It makes this assertion invalid in some cases. We should remove it.

I am not sure if removing it is the best course of action here (instead of expanding it). Also whatever we end up doing in containingBlockLogicalWidthForPositioned should also be applied to containingBlockLogicalHeightForPositioned.

obyknovenius committed:
ASSERTION FAILED: containingBlock.isInFlowPositioned() while running `the-anchor-attribute-003-crash.tentative.html` WPT Test
https://bugs.webkit.org/show_bug.cgi?id=267993

Reviewed by NOBODY (OOPS!).

According to the spec [1], some properties, such as "position",
"transform", "will-change", "contain", can cause a box to establish an
absolute positioning containing block and become a containing block for
an absolute positioned box.

It makes this assertion invalid in some cases. We should remove it.

The similar change was done in Blink in [2] where this test was
introduced.

1. https://drafts.csswg.org/css-position/#absolute-positioning-containing-block
2. https://bugs.chromium.org/p/chromium/issues/detail?id=1486148

* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::containingBlockLogicalHeightForPositioned const):
